cocos/core/data/decorators/property.ts

```diff
@@ -248,9 +248,12 @@ function setDefaultValue<T> (
     propertyKey: PropertyKey,
     descriptorOrInitializer: BabelPropertyDecoratorDescriptor | Initializer | undefined,
 ) {
-    if (descriptorOrInitializer) {
+    if (descriptorOrInitializer !== undefined) {
         if (typeof descriptorOrInitializer === 'function') {
             propertyStash.default = getDefaultFromInitializer(descriptorOrInitializer);
+        } else if (descriptorOrInitializer === null) {
+            // some decorated properties we haven't specified default value, then the initializer should be null.
```

Comment:
// If the `descriptorOrInitializer === null`, means this decorator is handled by babel but there is no initiailizer specified.